src/lang/eval/EvalUtil.ml

```diff
@@ -120,7 +120,7 @@ module Configuration = struct
     (* Current environment parameters and local variables *)
     env : Env.t;
     (* Contract fields *)
-    fields : (string * literal) list;
```

Comment:
This is odd. Did we actually not use the literals, only their types?

Reply:
No. we used literals directly earlier. Now it will come from the `StateService` "oracle".

Reply:
We don't need `typ` here, strictly. That's just there for convenience. So now, state values are not part of the execution environment, but are fetched on demand (see how loads and stores are handled).